app/controllers/SubmissionsController.scala

```diff
@@ -64,10 +65,10 @@ class SubmissionsController @Inject()(
     }
   }
 
-  def amend(id: String): Action[AnyContent] = authenticate.async { implicit request =>
+  def amend(id: String): Action[AnyContent] = (authenticate andThen verifyEmail).async { implicit request =>
     val redirect = Redirect(controllers.declaration.routes.SummaryController.displayPage(Mode.Amend))
 
-    val actualDeclaration: Future[Option[ExportsDeclaration]] = request.declarationId.map { decId =>
+    val actualDeclaration: Future[Option[ExportsDeclaration]] = request.request.declarationId.map { decId =>
```

Comment:
I'd suggest not to create this kind of caterpillars (`request.request.req...`). If this value is required, then a simple method in `VerifiedEmailRequest` could make it shorter.

Reply:
:+1: